Feat: Update font

truph01 opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->

### Fixed Issues
<!---
1. Please postfix `$` with a URL link to the GitHub issue this Pull Request is fixing. For example, `$ https://github.com/Expensify/App/issues/<issueID>`.
2. Please postfix  `PROPOSAL:` with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).  For example, `PROPOSAL: https://github.com/Expensify/App/issues/<issueID>#issuecomment-1369752925`

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking and its automation will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<issueID>
$ https://github.com/Expensify/App/issues/<issueID(comment)>

Do NOT only link the issue number like this: $ #<issueID>
--->
$ https://github.com/Expensify/App/issues/43512
PROPOSAL: https://github.com/Expensify/App/issues/43512#issuecomment-2162284229


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

- [x] Verify that no errors appear in the JS console

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

- [x] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/170341821/b8cb0f1d-0cf1-4285-a7e2-5b09a90fb9a7


</details>

<details>
<summary>Android: mWeb Chrome</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/170341821/0695e6b5-2181-49f7-87e6-9bcdc9110c35


</details>

<details>
<summary>iOS: Native</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/170341821/764a4a0f-5988-48c9-af42-6f2d1357223d


</details>

<details>
<summary>iOS: mWeb Safari</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/170341821/9e27069f-85ba-4c68-9739-af52e1853a91


</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

<!-- add screenshots or videos here -->
- Large screen:


https://github.com/Expensify/App/assets/170341821/127c8695-f8ee-4a9c-bfe6-0de5e2cd6792

- Small screen:


https://github.com/Expensify/App/assets/170341821/11245905-56a2-4e7f-a87c-7b9d6eef78c8


</details>

<details>
<summary>MacOS: Desktop</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/170341821/5b00eeea-6134-416a-b369-3603f4bb4bc4


</details>

truph01 committed:
Feat: Update font

truph01 committed:
Fix: use textMico

truph01 committed:
Fix: Lint

shawnborton commented on src/styles/index.ts:
```diff
@@ -405,7 +407,11 @@ const styles = (theme: ThemeColors) =>
             lineHeight: variables.lineHeightSmall,
         },
-
+        textBold: {
```
We should have existing styles for this, no?

truph01 replied:
Yes. We do not have this style before.

shawnborton commented on src/styles/index.ts:
```diff
@@ -446,8 +452,7 @@ const styles = (theme: ThemeColors) =>
 
-        textNewKansasNormal: {
-            fontFamily: FontUtils.fontFamily.platform.EXP_NEW_KANSAS_MEDIUM,
+        fontSizeNormal: {
```
Is this a new style?

truph01 replied:
Oh. There was a similar style named textNormal:
```
        textNormal: {
            fontSize: variables.fontSizeNormal,
        },
```
So I removed ```fontSizeNormal``` and use ```textNormal``` instead.

shawnborton commented:
On desktop, the To & From should not be in bold.

shawnborton commented:
Can you check the spacing here:
![CleanShot 2024-06-14 at 08 21 45@2x](https://github.com/Expensify/App/assets/2319350/9d22fcf9-6bc1-4923-ab12-88c5000d8e4a)

We just want everything there to have an 8px gap

truph01 committed:
Fix: Remove fontSizeNormal

truph01 committed:
Fix: Remove bold style in desktop

truph01 commented:
@shawnborton Reply your https://github.com/Expensify/App/pull/43733#issuecomment-2167293132:
- Yes. The gap between avatar and name is 8px:
![image](https://github.com/Expensify/App/assets/170341821/e692bab8-0846-4569-893c-db396346685e)

shawnborton commented:
> Yes. The gap between avatar and name is 8px:

Ah sorry, that comment was from the mobile view of the Search page.

shawnborton commented:
Can you make it so that these To/From columns on desktop vertically center everything? Right now it looks like it's aligned to the top and it feels a bit off:
![CleanShot 2024-06-14 at 18 53 13@2x](https://github.com/Expensify/App/assets/2319350/8d78494e-9b7b-486e-837e-6f3477ffc9be)

shawnborton commented:
We may need to make the date column slightly wider as a result of this. We currently have a date PR in review that also adds the ability to see the year in dates for expenses from prior years, so perhaps as soon as that is merged, we can adjust the column widths there.

shawnborton commented:
This isn't your fault but it's something I think we should fix: the small text in mobile view under the merchant has some weird alignment going on, I think in part because we are giving it a min-height of 20px:
![CleanShot 2024-06-14 at 19 00 45@2x](https://github.com/Expensify/App/assets/2319350/c2842119-b2dd-45b7-b748-e6bf8d7d2f4f)

Can we make it so that we just use the standard line-height of the text, which should be 14px since it's micro-sized font:
![CleanShot 2024-06-14 at 19 02 20@2x](https://github.com/Expensify/App/assets/2319350/7ac75723-f16b-4641-8c00-59d942893d8b)

And then we just vertically center everything? Basically removing the min-height in my first screenshot of this comment should fix that.

luacmartins reviewed (commented):
@truph01 please let me know once you're addressed Shawn's comments

truph01 committed:
Fix: Merge main

truph01 commented:
@luacmartins Sure. I am working on it. It is related to style fix, so I need to test on all platforms.

luacmartins commented:
Sounds good, thanks for the update!

truph01 committed:
Fix: Remove category min line height

truph01 commented:
@shawnborton 

> Can you make it so that these To/From columns on desktop vertically center everything? Right now it looks like it's aligned to the top and it feels a bit off:

- Currently, I do not have any solution for it. The parent's height is 21, and the child's height is 16, so the ```alignItem: 'center'``` or ```'alignSelf: 'center'``` do not work.

> We may need to make the date column slightly wider as a result of this. We currently have a date PR in review that also adds the ability to see the year in dates for expenses from prior years, so perhaps as soon as that is merged, we can adjust the column widths there.

- We already have a logic to handle it. If we display the year, the date column is grow up. This is that change: https://github.com/Expensify/App/pull/43413/files#diff-097e3d461183958f84757dc8be87e3f24b862654235a9b3f59f75ae74a738e1


> The small text in mobile view under the merchant has some weird alignment going on

- I fixed it by removing the min line height attribute:

![image](https://github.com/Expensify/App/assets/170341821/d0dab2c6-97c2-4624-98a7-68aaf9473945)

c3024 commented:
@truph01 Could you add `Tests` for all changes made for both large and small screens?

c3024 commented:
@shawnborton 

This font size for headers appears too small to me on Chrome on a 27 inch monitor.

![Screenshot 2024-06-18 at 7 39 23 PM](https://github.com/Expensify/App/assets/102477862/56bc4f76-6c0c-4e55-b3a5-fc3c9faef052)

Is this okay?

shawnborton commented:
Yeah, we don't want to change the font size for the column headers at all.

shawnborton commented:
> We already have a logic to handle it. 

I mean even when there is no year present, the column width was too skinny and it was cutting off the simple date. We will need to make it wider.

> Currently, I do not have any solution for it. The parent's height is 21, and the child's height is 16

To be clear, all text in the desktop view should now have a line height of our normal line height (it's 21 now but it's being fixed to go back to 20). So can we just fix that alignment but making sure we aren't setting 16 as the line height in the To/From column, or any other column for that matter?

shawnborton commented:
Just tested again, notice how even when no year is applied, you still get a cut off date column:
![CleanShot 2024-06-19 at 09 20 07@2x](https://github.com/Expensify/App/assets/2319350/fa58742e-5b4d-4b38-821d-7d03b2413921)

So we need to slightly increase that width.

shawnborton commented:
For my comment about the vertical alignment of the avatar + to/from name, maybe we don't worry about that for now because I think when we reduce our line height back to 20px, it will look a bit better?
![CleanShot 2024-06-19 at 09 22 41@2x](https://github.com/Expensify/App/assets/2319350/8aee60d8-256f-408f-95ec-2b3010bec84d)

shawnborton commented:
Or we could simply add a 1px top margin to the avatar wrapper, which does seem to balance it out:
![CleanShot 2024-06-19 at 09 24 47@2x](https://github.com/Expensify/App/assets/2319350/9a287432-99b7-4a1c-aa53-9448fa0e14bf)

shawnborton commented:
On mobile, seeing some weird alignment when there is a category and tag present:
![CleanShot 2024-06-19 at 09 25 32@2x](https://github.com/Expensify/App/assets/2319350/44717659-905d-4caa-bcdb-a8165b603846)

shawnborton commented:
Looks like there is still 4px gap here on mobile, when we want it all to be just 8px:
![CleanShot 2024-06-19 at 09 32 10@2x](https://github.com/Expensify/App/assets/2319350/c0ed5da3-cd44-4555-acc6-7a1f11e9f61c)

truph01 committed:
Fix: Merge main

truph01 committed:
Fix: Comment

truph01 commented:
> Just tested again, notice how even when no year is applied, you still get a cut off date column:

- I updated. Here is result:

![image](https://github.com/Expensify/App/assets/170341821/b24bffb9-6a6e-4096-a8c2-52e1faebdb07)

> On mobile, seeing some weird alignment when there is a category and tag present

- I updated. Here is result:

![image](https://github.com/Expensify/App/assets/170341821/d8a82c3c-dbb2-4e08-89eb-d258c0d72e11)

> Also while we are here, can we make it so that the icon + date is vertically centered too?

- I updated. Here is result:

![image](https://github.com/Expensify/App/assets/170341821/3f0134de-a13e-45e8-94a3-2b2d5fcdf5f0)


> Looks like there is still 4px gap here on mobile, when we want it all to be just 8px:

- I updated. Here is result:

![image](https://github.com/Expensify/App/assets/170341821/71267e12-3f4a-48e7-a635-802cd8d8580c)

shawnborton commented:
Lovely. Just a few more comments. 

On mobile, can we make it so that the smaller text uses the same baseline alignment?
![CleanShot 2024-06-19 at 10 51 26@2x](https://github.com/Expensify/App/assets/2319350/5dbd4752-a704-4080-a166-f2c83c3d8303)

Ideally the merchant and amount use the same font size and line height, and then both of those should have the same gap between them and the smaller text below.

Then a different comment - given that we are losing a lot of horizontal space with the updated font size, could we update the expense type column on desktop to be 20px wide? Basically the same exact width as the icon itself. We need to update that in the column header as well.

truph01 commented:
> Ideally the merchant and amount use the same font size and line height, and then both of those should have the same gap between them and the smaller text below.

@shawnborton Currently, the merchant has lineHeight: 21 and the amount has lineHeight: 20. So should I update the merchant's lineHeight to 20?

shawnborton commented:
You could make it so that they both use 20, yup.

truph01 committed:
Fix: Comment

truph01 commented:
@shawnborton I addressed your [comment](https://github.com/Expensify/App/pull/43733#issuecomment-2178130278)

shawnborton commented:
Nice! Going to re-build.

shawnborton commented on src/styles/utils/index.ts:
```diff
@@ -1559,7 +1559,7 @@ const createStyleUtils = (theme: ThemeColors, styles: ThemeStyles) => ({
                 break;
             case CONST.SEARCH_TABLE_COLUMNS.DATE:
-                columnWidth = getWidthStyle(shouldExtendDateColumn ? variables.w80 : variables.w44);
+                columnWidth = getWidthStyle(shouldExtendDateColumn ? variables.w80 : variables.w50);
```
FWIW, 80 isn't quite big enough when we have a year present:
![CleanShot 2024-06-19 at 16 40 39@2x](https://github.com/Expensify/App/assets/2319350/57d0319e-8f5c-44e8-b246-60798971f029)

And for the other case, instead of 50, can you try using 52 or 56?

luacmartins reviewed (commented):
@truph01 we're getting close! we have one more request from @shawnborton though

truph01 committed:
Fix: Update date width

truph01 commented:
@shawnborton I resolved your [comment](https://github.com/Expensify/App/pull/43733#discussion_r1646328296)

shawnborton commented:
Sweet, let me rebuild again.

luacmartins reviewed (changes requested):
The total column shown for report transactions has a smaller font:

![Screenshot 2024-06-20 at 2 13 43 PM](https://github.com/Expensify/App/assets/22219519/c9cdafdb-63e5-4f25-b535-b96ea806d4db)

shawnborton commented:
Good catch!

Also wanted to bump @Expensify/design to check this out - I'm wondering if the lack of bold letters for the To/From actually looks way worse on mobile?
![CleanShot 2024-06-20 at 22 34 32@2x](https://github.com/Expensify/App/assets/2319350/74cd3e0a-30d2-45bb-9931-48ce4447e136)

truph01 committed:
Fix: Merge main

truph01 committed:
Fix: Incorrect total amount font

truph01 commented:
> The total column shown for report transactions has a smaller font:

- @luacmartins I fixed your comment. This is result:
![image](https://github.com/Expensify/App/assets/170341821/d7b62b2e-d64f-4709-8964-0fa70e11d2b8)

truph01 commented:
Just need to wait for the @Expensify/design's response in https://github.com/Expensify/App/pull/43733#issuecomment-2181512414

truph01 committed:
Fix lint

truph01 committed:
Fix: Lint

shawnborton commented:
Cool, let's wait until they chime in today.

dubielzyk-expensify commented:
> Also wanted to bump @Expensify/design to check this out - I'm wondering if the lack of bold letters for the To/From actually looks way worse on mobile?

I don't feel super strongly here so happy to lean on you both. Unbolded saves a few precious pixels, but bolded makes it stand out a bit more against the rest of the regular text. I probably lean slightly bold, but again, not super strongly.

shawnborton commented:
If we put them back to bold, we get something like this:
![CleanShot 2024-06-21 at 08 13 19@2x](https://github.com/Expensify/App/assets/2319350/c1fd602d-8e8b-409d-9b55-5b3458c4a3a8)

Which I think I like better, as it has some harmony with the button or badge on the right side which uses bold weight as well.

shawnborton commented:
Another small thing I am noticing: it looks like the expense cards are only 100px tall:
![CleanShot 2024-06-21 at 08 14 30@2x](https://github.com/Expensify/App/assets/2319350/ecf8a453-7ee4-4eba-8336-0a88966e1043)

Whereas in Figma, we have them at 108:
![CleanShot 2024-06-21 at 08 15 08@2x](https://github.com/Expensify/App/assets/2319350/87598fd2-d4f6-4775-b04b-38cffd828609)

I think that's because this top header wrapper here only has a margin-bottom of 8px:
![CleanShot 2024-06-21 at 08 15 44@2x](https://github.com/Expensify/App/assets/2319350/b351477a-3511-4a10-a4bf-f9ce98596165)

Whereas it should actually have a margin bottom of 16px:
![CleanShot 2024-06-21 at 08 16 40@2x](https://github.com/Expensify/App/assets/2319350/1238c88e-41f7-48fb-adb8-419795895194)


Can you make that change as well please? Thanks!

truph01 committed:
Fix: Using margin bottom 16

truph01 commented:
> Whereas it should actually have a margin bottom of 16px:

- @shawnborton I fixed it. This is result:
<img width="326" alt="Screenshot 2024-06-21 at 16 30 02" src="https://github.com/Expensify/App/assets/170341821/511a1549-645a-4199-8204-792634f7a89f">

> Also wanted to bump @Expensify/design to check this out - I'm wondering if the lack of bold letters for the To/From actually looks way worse on mobile?

- What is the final decision on this?

shawnborton commented:
I think we should do bold names, so let's implement that quickly and then test again. Thanks!

truph01 committed:
Fix: Add bold style

truph01 commented:
> I think we should do bold names, so let's implement that quickly and then test again. Thanks!

- Here is the result, the font-weight is 700:
<img width="326" alt="Screenshot 2024-06-21 at 16 58 46" src="https://github.com/Expensify/App/assets/170341821/ff516290-cee2-4762-8128-2e9107e657ac">

shawnborton commented:
Looking good to me! Let's get this into final review.

c3024 commented:
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Android: mWeb Chrome</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: Native</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: mWeb Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Desktop</summary>

<!-- add screenshots or videos here -->

</details>

luacmartins reviewed (approved):
LGTM

luacmartins commented:
@c3024 @shawnborton all yours

c3024 reviewed (approved):
LGTM too!

dannymcclain commented:
Sorry meant to weigh in on this before, but I'm into the bold for names on mobile!

luacmartins commented:
Chatted 1:1 with @shawnborton and we're good to merge!

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/9619839966) to staging by https://github.com/luacmartins in version: 9.0.1-0 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

shawnborton commented:
Sorry for the late feedback, but @truph01 I found a case where the date column isn't wide enough and causes truncation:
![CleanShot 2024-06-24 at 11 18 18@2x](https://github.com/Expensify/App/assets/2319350/884fdcd3-8039-4021-9090-f3dc24371cc7)

Can you follow up with a quick fix for that?

truph01 commented:
@shawnborton Sure. I wonder which value should we use?https://github.com/Expensify/App/blob/2026f62b5074d892163f837ca7b030b84c8e350f/src/styles/utils/index.ts#L1562

shawnborton commented:
I think just bumping to 88 might do the trick, but let's check a variety of dates to make sure. Thanks @truph01 !

luacmartins commented:
It seems like we need at least 92px. The date `May 30, 2030` or `May 30, 2000` gets truncated otherwise.

luacmartins commented:
PR [here](https://github.com/Expensify/App/pull/44292)

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/9668528895) to production by https://github.com/yuwenmemon in version: 9.0.1-19 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅
